File: src/main/java/com/piveguyz/empickbackend/member/command/application/dto/MemberSignUpRequestDTO.java
package com.piveguyz.empickbackend.member.command.application.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
public class MemberSignUpRequestDTO {

    @NotBlank
    private String password;

    @NotBlank
    private String name;

    private String birth;

    @NotBlank
    private String phone;

    @NotBlank
    private String pictureUrl;

    @NotBlank
    private String email;

    @NotBlank
    private String address;

    private Integer vacationCount = 0;

    @NotNull
    private LocalDateTime hireAt;

    private LocalDateTime resignAt;

    private Integer deletedMemberId;
    private Integer updatedMemberId;
    private LocalDateTime lastLoginAt;

    @NotNull
    private Integer status = 1;

    private Integer departmentId;
    private Integer positionId;
    private Integer jobId;
    private Integer rankId;
}

File: src/dto/MemberSignUpRequestDTO.js
export default class MemberSignUpRequestDTO {
    constructor({
        password = '',
        name = '',
        birth = '',
        phone = '',
        pictureUrl = '',
        email = '',
        address = '',
        vacationCount = 0,
        hireAt = '',
        resignAt = '',
        deletedMemberId = 0,
        updatedMemberId = 0,
        lastLoginAt = '',
        status = 1,
        departmentId = null,
        positionId = null,
        jobId = null,
        rankId = null,
    } = {}) {
        this.password = password;
        this.name = name;
        this.birth = birth;
        this.phone = phone;
        this.pictureUrl = pictureUrl;
        this.email = email;
        this.address = address;
        this.vacationCount = vacationCount;
        this.hireAt = hireAt;
        this.resignAt = resignAt;
        this.deletedMemberId = deletedMemberId;
        this.updatedMemberId = updatedMemberId;
        this.lastLoginAt = lastLoginAt;
        this.status = status;
        this.departmentId = departmentId;
        this.positionId = positionId;
        this.jobId = jobId;
        this.rankId = rankId;
    }
} 